src/agent/scm/lib/svnwrapper.ts

```diff
@@ -0,0 +1,475 @@
+            (ret) => {
+                deferred.resolve(ret);
+            },
+            (err) => {
+                deferred.reject(err);
+            }
+        );
+        
+        return deferred.promise;
+    }
+
+    public switch(svnModule: SvnMappingDetails): Q.Promise<number> {
+        var deferred = Q.defer<number>();
+        
+        this.ctx.info("Switching " + svnModule.localPath
+                    + " to ^" + svnModule.serverPath
+                    + " with depth: " + svnModule.depth 
+                    + ", revision: " + svnModule.revision 
+                    + ", ignore externals: " + svnModule.ignoreExternals);
+        var args: string[] = [svnModule.serverPath,
+                              svnModule.localPath, 
+                             '--revision', svnModule.revision, 
+                             '--depth', this._toSvnDepth(svnModule.depth)];
+        if (svnModule.ignoreExternals) {
+            args.push('--ignore-externals');
+        }
+        
+        this._exec('switch', args)
+        .then(
+            (ret) => {
+                deferred.resolve(ret);
+            },
+            (err) => {
+                deferred.reject(err);
+            }
+        );
+        
+        return deferred.promise;
+    }
+
+    public checkout(svnModule: SvnMappingDetails): Q.Promise<number> {
+        var deferred = Q.defer<number>();
+        
+        this.ctx.info("Checking out " + svnModule.localPath
+                    + " with depth: " + svnModule.depth 
+                    + ", revision: " + svnModule.revision 
+                    + ", ignore externals: " + svnModule.ignoreExternals);
+
+        var args: string[] = [svnModule.serverPath,
+                              svnModule.localPath, 
+                             '--revision', svnModule.revision, 
+                             '--depth', this._toSvnDepth(svnModule.depth)];
+        if (svnModule.ignoreExternals) {
+            args.push('--ignore-externals');
+        }
+        
+        this._exec('checkout', args)
+        .then(
+            (ret) => {
+                deferred.resolve(ret);
+            },
+            (err) => {
+                deferred.reject(err);
+            }
+        );
+        
+        return deferred.promise;
+    }
+
+    public buildSvnUrl(sourceBranch: string, serverPath?: string): string {
+        var url: string = this.endpoint.url.replace('\\', '/');
+        
+        if ((url == null) || (url.length == 0)) {
+            throw new Error("Connection endpoint URL cannot be empty.")
+        }
+        
+        url = this.appendPath(url, sourceBranch);
+        if (serverPath) {
+            url = this.appendPath(url, serverPath);
+        }
+
+        return url;
+    }
+    
+    public appendPath(base: string, path: string): string {
+        var url = base.replace('\\', '/');
+
+        if (path && (path.length > 0)) {
+            var matches: RegExpMatchArray = url.match(/.*\/$/);
```

Comment:
Does this need to be a regex? We're just looking to see whether the string ends with a / so I think we can just check path[path.length - 1] == '/' ... not that it makes a big difference, but I think it would be easier to read.